### PR TITLE
Warn that swiftlint --fix can break compilation on double-optionals

### LINKED ISSUE #127: setup-linters: SwiftLint redundant_nil_coalescing auto-fix can break double-optional code
## Problem

The `setup-linters` workflow runs `swiftlint --fix` during the initial lint pass. The `redundant_nil_coalescing` rule auto-fix removes `?? nil` expressions, but this is incorrect when the expression involves a double-optional (`T??`). Removing `?? nil` from `dictionary[key] ?? nil` where the dictionary's value type is `T?` changes the return type from `T?` to `T??`, causing a compilation error.

## What happened

A mock pasteboard in tests had:

```swift
// dataStore: [NSPasteboard.PasteboardType: Data?]
func data(forType type: NSPasteboard.PasteboardType) -> Data? {
    dataStore[type] ?? nil  // Flattens Data?? to Data?
}
```

`swiftlint --fix` removed the `?? nil`, producing:

```swift
func data(forType type: NSPasteboard.PasteboardType) -> Data? {
    dataStore[type]  // Returns Data??, compilation error
}
```

This caused `swift test` to fail with: `value of optional type 'Data??' must be unwrapped to a value of type 'Data?'`

## Resolution applied

Restored the `?? nil` with an inline disable comment:

```swift
// swiftlint:disable:next redundant_nil_coalescing
dataStore[type] ?? nil // Flattens Data?? to Data?
```

## Suggested fix

This is ultimately a SwiftLint bug (the auto-fix doesn't account for double-optionals). However, the `setup-linters` skill should add a cautionary note to the workflow:

> After running `swiftlint --fix`, always re-run `swift build` or `swift test` to verify the auto-fixes didn't introduce compilation errors. Some auto-fix rules (notably `redundant_nil_coalescing`) can produce incorrect code in edge cases.

This could go in the Swift language reference and/or as a general note in the skill's error handling section.

### PR DESCRIPTION
**Pull request by cboone**
## Summary

- Add a cautionary note in the Swift language reference explaining that SwiftLint's `redundant_nil_coalescing` auto-fix can break double-optional code (`[K: V?]` lookups), with the recommended `// swiftlint:disable:next` workaround.
- Add `swift build` to the suggested `lint-fix` Makefile target so auto-fix breakage is caught locally rather than in CI, and update the lint-fix verification note to match.
- Add a generalized "Auto-fix introduces compilation errors" bullet to the skill's error handling section pointing readers at the per-language verification step.

## Test plan

- [ ] Read `plugins/setup-linters/skills/setup-linters/references/languages/swift.md` and confirm the new `swift build` step and the two notes (lint-fix verification pattern, auto-fix can break compilation) read clearly.
- [ ] Confirm the new bullet in `plugins/setup-linters/skills/setup-linters/SKILL.md` Error Handling reads as a general principle rather than a Swift-specific note.
- [ ] `yarn lint:fix && yarn validate` pass cleanly.
- [ ] `setup-linters` version bumped from `1.9.0` to `1.9.1` in both `plugins/setup-linters/.claude-plugin/plugin.json` and `.claude-plugin/marketplace.json`; marketplace `metadata.version` unchanged.

## Closes

Closes #127

### CODE REVIEW

File: plugins/setup-linters/skills/setup-linters/references/languages/swift.md [hunk 1]
```diff
@@ -78,8 +78,8 @@ Adjust to match the project's minimum Swift version.
 # SwiftLint (lint)
 swiftlint lint --strict
 
-# SwiftLint (auto-fix then verify)
-swiftlint lint --fix && swiftlint lint --strict
+# SwiftLint (auto-fix, verify lint, then verify compilation)
+swiftlint lint --fix && swiftlint lint --strict && swift build
```
[the remark this answers is not in the document]
Comment (cboone) [reply plugins/setup-linters/skills/setup-linters/references/languages/swift.md]: Good catch. Replaced `swift build` with `swift test --build-tests` in both the command-line example and the `lint-fix` Makefile target, and updated the surrounding notes to explain why (test targets are exactly where the `redundant_nil_coalescing` double-optional case showed up). Fixed in 6ae5f6f.

File: plugins/setup-linters/skills/setup-linters/references/languages/swift.md [hunk 2]
```diff
@@ -101,6 +101,7 @@ lint-fix: ## Auto-fix then verify
 	swiftlint lint --fix
 	swiftlint lint --strict
 	swiftformat .
+	swift build
```
[the remark this answers is not in the document]
Comment (cboone) [reply plugins/setup-linters/skills/setup-linters/references/languages/swift.md]: Same fix as the sibling thread on the command example: the `lint-fix` Makefile target now ends with `swift test --build-tests` so tests are compiled too. Fixed in 6ae5f6f.

File: plugins/setup-linters/skills/setup-linters/SKILL.md [hunk 3]
```diff
@@ -203,6 +203,7 @@ Ask whether to commit the setup. Suggest `chore: set up <tool list>` as the comm
 - **Package manager not found**: If a Node.js project has no lockfile, default to npm. For non-Node projects, guide installation via Homebrew, Cargo, or the appropriate system package manager.
 - **Install failure**: If a tool fails to install, report the error and continue with the next tool.
 - **Config conflict**: If a generated config would overwrite an existing file, ask before overwriting.
+- **Auto-fix introduces compilation errors**: Some linter auto-fix rules can produce incorrect code in edge cases (notably SwiftLint's `redundant_nil_coalescing` rule, which can break double-optional expressions). After any auto-fix pass, re-run the project's compiler or test command (`swift build`, `cargo build`, `go build`, `tsc`, etc.) to verify that the auto-fixes did not introduce compilation errors. The language-specific reference files include this verification step in their suggested `lint-fix` Makefile targets.
```
[the remark this answers is not in the document]
Comment (cboone) [reply plugins/setup-linters/skills/setup-linters/SKILL.md]: Fair point. Reworded the sentence to "Where applicable, include a compile or test verification step in the language reference's suggested auto-fix target," which avoids implying that every language reference already has a `lint-fix` target. Fixed in 6ae5f6f.

**Comment by cboone** [comment]
## Copilot Feedback Resolved

Addressed 3 Copilot review comment(s) with code changes (commit 6ae5f6f):

| File                                                                | Category | Action                                                                                            |
| ------------------------------------------------------------------- | -------- | ------------------------------------------------------------------------------------------------- |
| `plugins/setup-linters/.../languages/swift.md:82`                   | Valid    | Replaced `swift build` with `swift test --build-tests` in command example so tests compile too    |
| `plugins/setup-linters/.../languages/swift.md:104`                  | Valid    | Same fix applied to the `lint-fix` Makefile target                                                |
| `plugins/setup-linters/skills/setup-linters/SKILL.md:206`           | Valid    | Reworded "language references include..." to avoid overclaiming about other language reference files |
